source/common/http/matching/data_impl.h

```diff
@@ -13,6 +13,9 @@ namespace Matching {
  */
 class HttpMatchingDataImpl : public HttpMatchingData {
 public:
+  explicit HttpMatchingDataImpl(const Network::ConnectionInfoProvider& connection_info_provider)
+      : connection_info_provider_(connection_info_provider) {}
```

Comment:
Do you think it would be better to provide a `StreamInfo` directly for HTTP matching data, compared to `ConnectionSocket` for network matching data?

Reply:
`StreamInfo` is complicated, it can be nested for internal redirects so I wanted to avoid bringing all that complexity implicitly.
I think we'll probably end-up using it though.